### ISSUE TITLE
ScatterView Test can't be instantiated for LayoutLeft

### ISSUE THREAD

**Issue by crtrott** [kokkos/kokkos#2168]
Getting this error:
```
In file included from /ascldap/users/crtrott/Kokkos/kokkos/containers/unit_tests/TestScatterView.hpp:47:0,
                 from /ascldap/users/crtrott/Kokkos/kokkos/containers/unit_tests/openmp/TestOpenMP_ScatterView.cpp:46:
/ascldap/users/crtrott/Kokkos/kokkos/containers/src/Kokkos_ScatterView.hpp: In instantiation of ‘void Kokkos::Experimental::contribute(Kokkos::View<DT1, VP ...>&, const Kokkos::Experimental::ScatterView<DT2, LY, ES, OP, CT, DP>&) [with DT1 = double**; DT2 = double* [3]; LY = Kokkos::LayoutLeft; ES = Kokkos::OpenMP; int OP = 0; int CT = 1; int DP = 0; VP = {Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u>}]’:
/ascldap/users/crtrott/Kokkos/kokkos/containers/unit_tests/TestScatterView.hpp:375:19:   required from ‘void Test::test_scatter_view_config<ExecSpace, Layout, duplication, contribution, op>::run_test(int) [with ExecSpace = Kokkos::OpenMP; Layout = Kokkos::LayoutLeft; int duplication = 1; int contribution = 0; int op = 0]’
/ascldap/users/crtrott/Kokkos/kokkos/containers/unit_tests/TestScatterView.hpp:396:5:   required from ‘Test::TestDuplicatedScatterView<ExecSpace, ScatterType>::TestDuplicatedScatterView(int) [with ExecSpace = Kokkos::OpenMP; int ScatterType = 0]’
/ascldap/users/crtrott/Kokkos/kokkos/containers/unit_tests/TestScatterView.hpp:455:62:   required from ‘void Test::test_scatter_view(int) [with ExecSpace = Kokkos::OpenMP; int ScatterType = 0]’
/ascldap/users/crtrott/Kokkos/kokkos/containers/unit_tests/TestScatterView.hpp:460:73:   required from here
/ascldap/users/crtrott/Kokkos/kokkos/containers/src/Kokkos_ScatterView.hpp:1265:3: error: no matching function for call to ‘Kokkos::Experimental::ScatterView<double* [3], Kokkos::LayoutLeft, Kokkos::OpenMP, 0, 1, 0>::contribute_into(Kokkos::View<double**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>, Kokkos::MemoryTraits<0u> >&) const’
   src.contribute_into(dest);
   ^~~
/ascldap/users/crtrott/Kokkos/kokkos/containers/src/Kokkos_ScatterView.hpp:1053:8: note: candidate: template<class ... RP> void Kokkos::Experimental::ScatterView<DataType, Kokkos::LayoutLeft, ExecSpace, Op, 1, contribution>::contribute_into(const Kokkos::View<DataType, RP ...>&) const [with RP = {RP ...}; DataType = double* [3]; int Op = 0; ExecSpace = Kokkos::OpenMP; int contribution = 0]
   void contribute_into(View<DataType, RP...> const& dest) const
        ^~~~~~~~~~~~~~~
/ascldap/users/crtrott/Kokkos/kokkos/containers/src/Kokkos_ScatterView.hpp:1053:8: note:   template argument deduction/substitution failed:
/ascldap/users/crtrott/Kokkos/kokkos/containers/src/Kokkos_ScatterView.hpp:1265:3: note:   mismatched types ‘double* [3]’ and ‘double**’
   src.contribute_into(dest);
```

After modifying the test file like this:
```diff
diff --git a/containers/unit_tests/TestScatterView.hpp b/containers/unit_tests/TestScatterView.hpp
index 2e16a89..53464bf 100644
--- a/containers/unit_tests/TestScatterView.hpp
+++ b/containers/unit_tests/TestScatterView.hpp
@@ -387,8 +387,13 @@ struct TestDuplicatedScatterView {
     test_scatter_view_config<ExecSpace, Kokkos::LayoutRight,
       Kokkos::Experimental::ScatterDuplicated,
       Kokkos::Experimental::ScatterNonAtomic, 
-      ScatterType> test_sv_config;
-    test_sv_config.run_test(n);
+      ScatterType> test_sv_right_config;
+    test_sv_right_config.run_test(n);
+    test_scatter_view_config<ExecSpace, Kokkos::LayoutLeft,
+      Kokkos::Experimental::ScatterDuplicated,
+      Kokkos::Experimental::ScatterNonAtomic, 
+      ScatterType> test_sv_left_config;
+    test_sv_left_config.run_test(n);
   }
 };
```

**Comment by crtrott** [comment]
Have a fix for that.

**Comment by crtrott** [comment]
This does not fix #2165
